### PR TITLE
feat(ras): custom newsletter list selection on registration

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Implements a RAS advanced setting for selecting custom newsletter lists, their order and default `checked` state when rendering in the registration modal.

image

### How to test the changes in this Pull Request:

1. Check out this branch and make sure you have RAS enabled and Newspack Newsletters connected with an ESP and enabled with subscription lists
2. Navigate to Newspack -> Engagement -> Reader Activation -> Advanced settings
3. Scroll to the "Newsletter Subscription Lists" section and toggle the "Custom newsletter lists on registration"
4. Confirm you can click on multiple available lists
5. Sort by clicking on the chevrons, make some checked by default and others not and save
6. Visit your site anonymously, click "Sign in" -> "I don't have an account" and confirm the configured lists and rendered in the defined order with the defined `checked` state

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/components/src/sortable-newsletter-list-control/style.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
+.newspack__newsletter-list-control {
+	&__selected {
+		.newspack-card {
+			display: flex;
+			// flex-direction: column;
```
Comment (adekbadek) [inline assets/components/src/sortable-newsletter-list-control/style.scss:5]: Commented out code should be removed.

**Review by adekbadek — CHANGES_REQUESTED** [review]
Something's off with the sorting UI. The chevron icons are removing the items instead of updating their position:

Screen Capture on 2023-10-19 at 15-54-46

Also, the setting is not saved correctly. The value is a mix of integers and arrays:

image

File: assets/components/src/sortable-newsletter-list-control/style.scss (hunk 1 above)
Comment (miguelpeixe) [reply assets/components/src/sortable-newsletter-list-control/style.scss]: Fixed in 90e05a5a4c8edf4d38951ef1391d6ac278cb4f2e

**Comment by miguelpeixe** [comment]
Thank you for catching that! It was an oversight when ddad1f5a11d95fae33d4f150a4100048468aa23b was introduced. Fixed in d4414c6b6cd64790008932b306c4dced11609912.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**